backend/apps/portfolio/tasks.py

```
@@ -0,0 +1,93 @@
+        portfolio.file.save(filename, file_content)
+        
+        portfolio.status = GeneratedPortfolio.Status.COMPLETED
+        portfolio.save(update_fields=["status", "file"])
+        
+        # Send a real-time notification to the user
+        from apps.notifications.services import NotificationService
+        NotificationService.send_in_app_notification(
+            user_id=user.id,
+            notification_type="SYSTEM",
+            title="Portfolio Ready",
+            message=f"Your developer portfolio has been successfully generated.",
```

Ruff (0.15.20)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)